JK Hera/src/main/java/ayds/hera/songinfo/moredetails/fulllogic/OtherInfoWindow.kt

```diff
@@ -0,0 +1,210 @@
+        contentPane.layout = BoxLayout(contentPane, BoxLayout.PAGE_AXIS)
+    }
+
+    private fun addPosterImageLabelTitle(container: Container) {
+        val label = JLabel()
+        label.text = POSTER_IMAGE_LABEL_TITLE
+        container.add(label)
+    }
+
+    private fun addImagePanel(container: Container) {
+        imagePanel = JPanel()
+        container.add(imagePanel)
+    }
+
+    private fun addPosterImageLabel(container: Container) {
+        posterImageLabel = JLabel()
+        container.add(posterImageLabel)
+    }
+
+    private fun addScrollPane(container: Container) {
+        scrollPane = JScrollPane(
+            descriptionPane,
+            JScrollPane.VERTICAL_SCROLLBAR_AS_NEEDED,
+            JScrollPane.HORIZONTAL_SCROLLBAR_AS_NEEDED
+        )
+        container.add(scrollPane)
+    }
+
+    private fun addDescriptionPanel() {
+        descriptionPane = JPanel()
+    }
+
+    private fun addTextPane(container: Container) {
+        textPane2 = JTextPane()
+        textPane2.isEditable = false
+        textPane2.contentType = "text/html"
+        textPane2.maximumSize = Dimension(600, 400)
+        container.add(textPane2)
+    }
+
+    private fun addSearchButtonPane(container: Container) {
+        searchButtonPane = JPanel()
+        searchButtonPane.layout = BorderLayout()
+        searchButtonPane.maximumSize = Dimension(600, 12)
+        container.add(searchButtonPane)
+    }
+
+    private fun addFullArticleButton(container: Container) {
+        fullArticleButton = JButton(VIEW_FULL_ARTICLE_BUTTON_TITLE)
+        container.add(fullArticleButton)
+    }
+
+    private fun initFrame() {
+        frame = JFrame(WINDOW_TITLE)
+        frame.minimumSize = Dimension(600, 600)
+        frame.pack()
+        frame.isVisible = true
+    }
+
+    private fun setImageNYTimesCover() {
+        try {
+            posterImageLabel.icon = ImageIcon(ImageIO.read(URL(NYT_IMAGE_URL)))
+        } catch (e: Exception) {
+        }
+    }
+
+    private fun initFullArticleListener(url: JsonElement) {
+        fullArticleButton.addActionListener { browseURL(url) }
+    }
+
+    private fun getResponseArticleData(song: Song): JsonObject {
+        val callResponse: Response<String> = articleDataHandler.getArtistInfo(song.artistName).execute()
+        return getJsonObject(callResponse)
+    }
+
+    private fun getJsonObject(callResponse: Response<String>): JsonObject {
+        val gson = Gson()
+        val jobj = gson.fromJson(callResponse.body(), JsonObject::class.java)
+        return jobj["response"].asJsonObject
+    }
+
+    private fun getJsonElement(response: JsonObject, str: String): JsonElement {
+        return response["docs"].asJsonArray[0].asJsonObject[str]
+    }
+
+    private fun markAsAlreadySearched(text: String?): String {
+        return "[*]$text"
+    }
+
+    private fun replaceBars(abstract: JsonElement) = abstract.asString.replace("\\n", "\n")
+
+    private fun isInDataBase(text: String?) = text != null
```

Comment:
Ver si un string es distinto de null no es "ver si esta en la base de datos". Lo que pueden hacer en todo caso es agregar la linea anterior a esta llamada en este metodo.

```suggestion
    private fun isInDataBase(text: String?): Boolean {
            var text = dataBase.getInfo(song.artistName)
            return text != null
    }
```

Además text debería tener otro nombre más representativo